src/Controller/BlogPostController.php

```diff
@@ -0,0 +1,207 @@
+            );
+        }
+
+        if ($postEnquiry->getTimestamp() === null) {
+            return new Response(
+                json_encode(["success" => false, "msg" => "Missing timestamp text"], JSON_THROW_ON_ERROR),
+                400,
+                ['Content-Type' => 'application/json']
+            );
+        }
+
+        $post = $blogPostService->createBlogPostFromEnquiry($postEnquiry, $user);
+
+
+        $entityManager->persist($post);
+        $entityManager->flush();
+
+        return new Response(
+            json_encode(["success" => true], JSON_THROW_ON_ERROR),
+            200,
+            ['Content-Type' => 'application/json']
+        );
+    }
+
+
+    /**
+     * @Route("/blogposts/{blogID}", name="blog-posts-delete", methods={"DELETE"})
+     * @throws NotSupported
+     * @throws \JsonException
+     *
+     */
+    public function removeBlogPost(
+        EntityManagerInterface $entityManager,
+        int $blogID,
+    ):
+    Response {
+
+        /** @var BlogPostRepository $blogPostRepository */
+        $blogPostRepository = $entityManager->getRepository(BlogPost::class);
+
+        $post = $blogPostRepository->find($blogID);
+
+        if (!$post) {
+            throw $this->createNotFoundException(
+                'No post found for id' . $blogID
+            );
+        }
+
+        try {
+            $blogPostRepository->removeById($blogID);
+        } catch (OptimisticLockException|ORMException $e) {
+            return new Response(
+                json_encode(["success" => true, "msg" => $e->getMessage()], JSON_THROW_ON_ERROR),
+                200,
+                ['Content-Type' => 'application/json']
+            );
+
+        }
+
+        return new Response(
+            json_encode(["success" => true, "removedID" => $blogID], JSON_THROW_ON_ERROR),
+            200,
+            ['Content-Type' => 'application/json']
+        );
+    }
+
+    /**
+     * @Route("/blogposts/{blogID}", name="blog-posts-update", methods={"PUT"})
+     * @throws NotSupported
+     * @throws \JsonException
+     *
+     */
+    public function updateBlogPost(
+        Request $request,
+        EntityManagerInterface $entityManager,
+        DTOSerializer $serializer,
+        int $blogID,
+    ):
+    Response {
+        /** @var BlogPostEnquiry $postEnquiry */
+        $postEnquiry = $serializer->deserialize(
+            $request->getContent(),
+            BlogPostEnquiry::class,
+            'json'
+        );
+        /** @var BlogPostRepository $blogPostRepository */
+        $blogPostRepository = $entityManager->getRepository(BlogPost::class);
+        $blogPostService = new BlogPostService($blogPostRepository);
+
+        $updatedPost = $blogPostService->updateBlogPostFromEnquiry($postEnquiry, $blogID);
+
+
+        if (!$updatedPost) {
+            throw $this->createNotFoundException(
+                'No post found for id ' . $blogID
+            );
+        }
```

Comment:
(as mentioned by raj) Code flow: validations first (and throw the errors), before performing the operation/logic